src/module/item/effect/document.ts

```diff
@@ -186,6 +188,8 @@ class EffectPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Ab
 
                 const currentValue = Number(badgeChange.value ?? currentBadge.value ?? 1);
                 badgeChange.value = Math.clamped(currentValue, minValue, maxValue);
+
+                if (badgeChange["-=labels"] === null) badgeChange["loop"] = false;
```

Comment:
Move this to outside, like how min/max are. You also need to handle the case where a formerly counter badge with labels that looped is converted to another type. Also delete it instead of setting to false since its optional in the source data.